exchange/__main__.py

```diff
@@ -0,0 +1,104 @@
+            [
+                telebot.types.InlineKeyboardButton(
+                    text=rate.Cur_Name, callback_data=rate.Cur_Abbreviation
+                )
+            ]
+            for rate in rates
+        ]
+        keyboard = telebot.types.InlineKeyboardMarkup(buttons)
+        bot.send_message(
+            chat_id=message.chat.id, text=localization.local['select_country'], reply_markup=keyboard
+        )
+
+
+@bot.callback_query_handler(func=lambda message: True)
+@bot.message_handler(func=lambda message: True)
+def current_exrates_rates(message):
+    if isinstance(message, telebot.types.Message):
+        message_handler: telebot.types.Message = message
+        user = User(
+            user_id=message_handler.from_user.id,
+            first_name=message_handler.from_user.first_name,
+            last_name=message_handler.from_user.last_name,
+            html_text=message_handler.html_text,
+            date=message_handler.date,
+            chat_id=message_handler.chat.id,
+        )
+    else:
+        callback_handler: telebot.types.CallbackQuery = message
+        user = User(
+            user_id=callback_handler.from_user.id,
+            first_name=callback_handler.from_user.first_name,
+            last_name=callback_handler.from_user.last_name,
+            html_text=callback_handler.data,
+            date=callback_handler.message.date,
+            chat_id=callback_handler.message.chat.id,
+        )
+    try:
+        log.info('insert user data to DB...')
+        DBUserLog(db_connection).insert(user)
+        DBUser(db_connection).insert(user)      
+    except:
+        log.error(localization.local['errors']['system'])
+        bot.send_message(chat_id=user.chat_id, text=localization.local['errors']['system'])
+    try:
+        response = requests.get(
+            f'{config.EXCHANGE_API}/exrates/rates', params={'periodicity': 0}
+        )
+    except requests.ConnectionError:
+        log.error(localization.local['errors']['system'])
+        bot.send_message(chat_id=user.chat_id, text=localization.local['errors']['system'])
+    else:
+        if response.status_code != 200:
+            bot.send_message(
+                chat_id=user.chat_id, text=localization.local['errors']['system']
+            )
+        rates: list[ExratesRates] = [
+            rate
+            for rate in [ExratesRates(**rate) for rate in response.json()]
+            if rate.Cur_Abbreviation == str(user.html_text).upper()
+        ]
+        bot.send_message(
+                chat_id=user.chat_id,
+                text=localization.local['rate']
+                .format(
+                    rates[0].Cur_Name, 
+                    rates[0].Cur_Scale, 
+                    rates[0].Cur_OfficialRate
+                ),
+            ) if len(rates) == 1 else bot.send_message(
+                chat_id=user.chat_id,
+                text=localization.local['errors']['not_found'].format(user.html_text),
+            )
```

Comment:
По мне так лучше сделать явную проверку на плохой кейс и `return-early`, чем городить такой большой блок кода с тернарным оператором:

```suggestion
        if len(rates) == 1: 
            bot.send_message(
                chat_id=user.chat_id,
                text=localization.local['errors']['not_found'].format(user.html_text),
            )
            return

        bot.send_message(
            chat_id=user.chat_id,
            text=localization.local['rate']
            .format(
                rates[0].Cur_Name, 
                rates[0].Cur_Scale, 
                rates[0].Cur_OfficialRate
            ),
        )
```

Reply:
Исправил